egs/wsj/s5/steps/libs/nnet3/train/common.py

```diff
@@ -318,6 +318,222 @@ def get_learning_rate(iter, num_jobs, num_iters, num_archives_processed,
+            num_archives_to_process, this_dropout_str)
+        dropout_schedule.append((component_name, this_dropout_values))
+
+    logger.info("Dropout schedules for component names is as follows:")
+    logger.info("<component-name-pattern>: [(num_archives_processed), "
+                "(dropout_proportion) ...]")
+    for name, schedule in dropout_schedule:
+        logger.info("{0}: {1}".format(name, schedule))
+
+    return dropout_schedule
+
+
+def _parse_dropout_string(num_archives_to_process, dropout_str):
+    """Parses the dropout schedule from the string corresponding to a
+    single component in --trainer.dropout-schedule.
+
+    Arguments:
+        num_archives_to_process: the number of archives that will
+            be processed over the entire duration.
+            For frame-level training, this is typically the number of archives
+            in disk multiplied by (frames_per_eg * num_epochs).
+            For chunk training, this is typically the number of archives in
+            disk multiplied by num_epochs.
+        dropout_str: Specifies dropout schedule for a particular component
+            name pattern.
+            See help for the option --trainer.dropout-schedule.
+
+    Returns a list of (num_archives_processed, dropout_proportion) tuples
+    sorted in descending order of num_archives_processed.
+    The data fraction in dropout_str is converted into num_archives_processed.
+    A data fraction of 1 corresponds to num_archives_to_process i.e. all data.
+    """
+    dropout_values = []
+    parts = dropout_str.strip().split(',')
+
+    try:
+        if len(parts) < 2:
+            raise Exception("dropout proportion string must specify "
+                            "at least the start and end dropouts")
+
+        # Starting dropout proportion
+        dropout_values.append((0, float(parts[0])))
+        for i in range(1, len(parts) - 1):
+            value_x_pair = parts[i].split('@')
+            if len(value_x_pair) == 1:
+                # Dropout proportion at half of training
+                dropout_proportion = float(value_x_pair)
+                num_archives = int(0.5 * num_archives_to_process)
+            else:
+                assert len(value_x_pair) == 2
+
+                dropout_proportion = float(value_x_pair[0])
+                data_fraction = float(value_x_pair[1])
+                num_archives = round(float(data_fraction)
+                                     * num_archives_to_process)
+
+            if (num_archives < dropout_values[-1][0]
+                    or num_archives > num_archives_to_process):
+                logger.error(
+                    "Failed while parsing value %s in dropout-schedule. "
+                    "dropout-schedule must be in incresing "
+                    "order of data fractions.", value_x_pair)
+                raise ValueError
+
+            dropout_values.append(num_archives, float(dropout_proportion))
```

Comment:
@vimalmanohar there should be` dropout_values.append((num_archives, float(dropout_proportion)))`